File: app/src/main/java/gardezi/io/amiibofinder/dagger/components/AppComponent.java
package gardezi.io.amiibofinder.dagger.components;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import gardezi.io.amiibofinder.AmiiboApplication;
import gardezi.io.amiibofinder.dagger.modules.AppModule;
import gardezi.io.amiibofinder.dagger.modules.MainActivityModule;
import gardezi.io.amiibofinder.dagger.modules.NetworkModule;

@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, MainActivityModule.class, NetworkModule.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(AmiiboApplication application);
        AppComponent build();
    }

    void inject(AmiiboApplication application);
}

File: app/src/main/java/gardezi/io/amiibofinder/dagger/components/AppComponent.kt
package gardezi.io.amiibofinder.dagger.components

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import gardezi.io.amiibofinder.AmiiboApplication
import gardezi.io.amiibofinder.dagger.AppModule
import gardezi.io.amiibofinder.dagger.MainActivityModule
import gardezi.io.amiibofinder.dagger.modules.DataModule
import javax.inject.Singleton

@Singleton
@Component(modules = [(AndroidInjectionModule::class), (AppModule::class), (MainActivityModule::class), (DataModule::class)])
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: AmiiboApplication): Builder

        fun build(): AppComponent
    }

    fun inject(application: AmiiboApplication)
}
